### PR TITLE
flake8

### PR DESCRIPTION
**Pull request by oliver-sanders**
We used to have good code style checking / testing, however, with the rebranding of `pep8` to `pycodestyle` we lost most of this functionality.

This adds `flake8`, perhaps the must successful, certainly the lest controversial Python style/error checker to our dev process.

Fixes at least ~one~ two genuine bugs.

Assigning @hjoliver and @wxtim as between the three of us we are responsible for most of these violations 🤭.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Does not need tests (why?).
- [x] No change log entry required (why? e.g. invisible to users).
- [x] No documentation update required.
- [ ] Created an issue at [cylc-flow conda-forge repository](https://github.com/conda-forge/cylc-flow-feedstock) with version changes (if you changed dependencies in `setup.py`, see `recipe/meta.yaml`).

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -30,6 +30,7 @@ jobs:
       - name: Style
         run: |
           pycodestyle
+          flake8
```
Comment (oliver-sanders) [inline .github/workflows/test.yml:33]: `pycodestyle` keep or kill?

File: cylc/flow/network/graphql.py [hunk 2]
```diff
@@ -226,12 +225,22 @@ def args_selection_search(
 
 
 def execute_and_validate_and_strip(
-        schema,  # type: GraphQLSchema
-        document_ast,  # type: Document
-        *args,  # type: Any
-        **kwargs  # type: Any
```
Comment (oliver-sanders) [inline cylc/flow/network/graphql.py:232]: Flake8 expects these types to be imported? Simple solution move them to a Google-style docstring like we use elsewhere.

File: cylc/flow/scripts/cylc_check_software.py [hunk 3]
```diff
@@ -148,10 +148,11 @@ def check_py_module_ver(module, min_ver, write=True):
         if module == 'EmPy':
             # we want the 'em' module, but there is more than one out there
             # we want the one that provides 'Interpreter'
-            from em import Interpreter
+            from em import Interpreter  # noqa: F401
```
Comment (oliver-sanders) [inline cylc/flow/scripts/cylc_check_software.py:151]: Per-case ignores use `# noqa: <code>`

File: tox.ini [hunk 4]
```diff
@@ -11,3 +11,15 @@ exclude=
     __pycache__,
     .tox,
     **data_messages_pb2.py
+
+[flake8]
+exclude=
+    build,
+    dist,
+    tests,
```
Comment (oliver-sanders) [inline tox.ini:19]: Too much work right now but we should remove `tests` from the exclude list.

**Comment by hjoliver** [comment]
> between the three of us we are responsible for most of these violations

I resemble that remark.

File: .github/workflows/test.yml (hunk 1 above)
Comment (hjoliver) [reply .github/workflows/test.yml:33]: Does it do anything useful that flake8 doesn't?

File: cylc/flow/scripts/cylc_help.py [hunk 5]
```diff
@@ -451,7 +449,7 @@ def help_func():
                 raise SystemExit(exc2)
             else:
                 # cylc COMMAND --help
-                command_args = ['--help']
+                pass
```
Comment (hjoliver) [inline cylc/flow/scripts/cylc_help.py:452]: Just remove the `else:` block?

File: cylc/flow/scripts/cylc_trigger.py [hunk 6]
```diff
@@ -28,18 +28,8 @@
 
 """
 
-import re
-import os
-import time
-import shutil
-import difflib
-from subprocess import call
```
Comment (hjoliver) [inline cylc/flow/scripts/cylc_trigger.py:36]: (@dwsutherland just noticed these unused imports today)

File: cylc/flow/scripts/cylc_trigger.py [hunk 7]
```diff
@@ -28,18 +28,8 @@
...
-
-from cylc.flow.exceptions import CylcError, UserInputError
 from cylc.flow.option_parsers import CylcOptionParser as COP
 from cylc.flow.network.client import SuiteRuntimeClient
-from cylc.flow.cfgspec.glbl_cfg import glbl_cfg
-from cylc.flow.task_job_logs import JOB_LOG_DIFF
```
Comment (hjoliver) [inline cylc/flow/scripts/cylc_trigger.py:42]: (Actually `JOB_LOG_DIFF` could be removed from `cylc/flow/task_job_logs` now - it's not used anywhere).

File: cylc/flow/task_pool.py [hunk 8]
```diff
@@ -37,13 +37,22 @@
...
+    TASK_STATUS_EXPIRED,
+    TASK_STATUS_FAILED,
+    TASK_STATUS_QUEUED,
+    TASK_STATUS_READY,
+    TASK_STATUS_RETRYING,
+    TASK_STATUS_RUNNING,
+    TASK_STATUS_SUBMITTED,
+    TASK_STATUS_SUBMIT_RETRYING,
+    TASK_STATUS_SUCCEEDED,
+    TASK_STATUS_WAITING
```
Comment (hjoliver) [inline cylc/flow/task_pool.py:54]: (You can't blame flake8 for this reformatting! :grin: )

Comment (oliver-sanders) [reply cylc/flow/task_pool.py:54]: No that's just good coding :)

File: cylc/flow/scripts/cylc_trigger.py (hunk 7 above)
Comment (oliver-sanders) [reply cylc/flow/scripts/cylc_trigger.py:42]: Removed.

File: .github/workflows/test.yml (hunk 1 above)
Comment (hjoliver) [reply .github/workflows/test.yml:33]: https://pypi.org/project/flake8/

> Flake8 is a wrapper around these tools:
>
>-    PyFlakes
>-    pycodestyle
>-   Ned Batchelder’s McCabe script
> 
> Flake8 runs all the tools by launching the single flake8 command. It displays the warnings in a per-file, merged 
> output.
> 
> It also adds a few features: 
> ...

If that's true, there's no reason to keep pycodestyle as well.

File: tox.ini (hunk 4 above)
Comment (wxtim) [reply tox.ini:19]: Perhaps this should be a ticket?

**Pull request merged**